drivers/sensors/bme680.c

```diff
@@ -0,0 +1,1429 @@
+
+/****************************************************************************
+ * Name: bme680_get_calib_data
+ *
+ * Description:
+ *   Read sensor-specific parameters and store them for later
+ * use in computing the compensated values.
+ *
+ ****************************************************************************/
+
+static int bme680_get_calib_data(FAR struct bme680_dev_s *priv)
+{
+  uint8_t coeff[BME680_COEFF_SIZE];
+  int ret;
+
+  /* Get first part of the calibration data. */
+
+  ret = bme680_getregs(priv, BME680_COEFF_ADDR1, coeff,
+                       BME680_COEFF_ADDR1_LEN);
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  /* Concatenate the second part of the data to coeff */
+
+  ret = bme680_getregs(priv, BME680_COEFF_ADDR2,
+                       &coeff[BME680_COEFF_ADDR1_LEN],
+                       BME680_COEFF_ADDR2_LEN);
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  /* Get data */
+
+  priv->dev.calib.t1 = coeff[BME680_T1_MSB_REG] << 8
+                      | coeff[BME680_T1_LSB_REG];
+  priv->dev.calib.t2 = coeff[BME680_T2_MSB_REG] << 8
+                      | coeff[BME680_T2_LSB_REG];
+  priv->dev.calib.t3 = coeff[BME680_T3_REG];
+
+#ifndef CONFIG_BME680_DISABLE_PRESS_MEAS
+  priv->dev.calib.p1 = coeff[BME680_P1_MSB_REG] << 8
+                      | coeff[BME680_P1_LSB_REG];
+  priv->dev.calib.p2 = coeff[BME680_P2_MSB_REG] << 8
+                      | coeff[BME680_P2_LSB_REG];
+  priv->dev.calib.p3 = coeff[BME680_P3_REG];
+  priv->dev.calib.p4 = coeff[BME680_P4_MSB_REG] << 8
+                      | coeff[BME680_P4_LSB_REG];
+  priv->dev.calib.p5 = coeff[BME680_P5_MSB_REG] << 8
+                      | coeff[BME680_P5_LSB_REG];
+  priv->dev.calib.p6 = coeff[BME680_P6_REG];
+  priv->dev.calib.p7 = coeff[BME680_P7_REG];
+  priv->dev.calib.p8 = coeff[BME680_P8_MSB_REG] << 8
+                      | coeff[BME680_P8_LSB_REG];
+  priv->dev.calib.p9 = coeff[BME680_P9_MSB_REG] << 8
+                      | coeff[BME680_P9_LSB_REG];
+  priv->dev.calib.p10 = coeff[BME680_P10_REG];
+#endif /* !CONFIG_BME680_DISABLE_PRESS_MEAS */
+
+#ifndef CONFIG_BME680_DISABLE_HUM_MEAS
+  priv->dev.calib.h1 = (uint16_t) (((uint16_t) coeff[BME680_H1_MSB_REG] << 4)
+      | (coeff[BME680_H1_LSB_REG] & BME680_BIT_H1_DATA_MSK));
+  priv->dev.calib.h2 = (uint16_t) (((uint16_t) coeff[BME680_H2_MSB_REG] << 4)
+      | ((coeff[BME680_H2_LSB_REG]) >> 4));
+  priv->dev.calib.h3 = coeff[BME680_H3_REG];
+  priv->dev.calib.h4 = coeff[BME680_H4_REG];
+  priv->dev.calib.h5 = coeff[BME680_H5_REG];
+  priv->dev.calib.h6 = coeff[BME680_H6_REG];
+  priv->dev.calib.h7 = coeff[BME680_H7_REG];
+#endif /* !CONFIG_BME680_DISABLE_HUM_MEAS */
+
+#ifndef CONFIG_BME680_DISABLE_GAS_MEAS
+
+  /* Gas-related coefficients */
+
+  priv->dev.calib.gh1 = coeff[BME680_GH1_REG];
+  priv->dev.calib.gh2 = coeff[BME680_GH2_MSB_REG] << 8
+                      | coeff[BME680_GH2_LSB_REG];
+  priv->dev.calib.gh3 = coeff[BME680_GH3_REG];
+
+  uint8_t temp_val = 0;
```

Comment:
Please move variable declaration to the beginning of function to make C89 compatibility.